pkg/healthcheck/healthcheck_test.go

```diff
@@ -2395,7 +2395,7 @@ data:
 	}
 }
 
-func getFakeConfig(secretScheme string, schemeInConfig string, issuerCerts *issuercerts.IssuerCertData) []string {
+func getFakeConfig(secretScheme string, schemeInConfig string, issuerCerts *issuercerts.IssuerCertData, configAnchorsModifier func(string) string) []string {
```

Comment:
Passing in a function that modifies just the config anchors for just the configmap feels hacky and shoehorned in here.  We should consider coming back and cleaning up/refactoring these tests later.

Reply:
I agree. I dont like it. How about we do the following. I introduce one more modifier function that works on the value that is in the secret. This makes it a bit less akward (symmetry) and allows us to write tests for the reverse case (more tests never hurt)

Reply:
I think the problem is that we're trying to fit all of our different test cases into the same structure, even when that's not a good fit.  This leads us to passing in the ability to do arbitrary transformations on various pieces.  It's not necessary that all tests fit into a single "table".  We can split out tests that have a different structure (e.g. where the configmap trust roots and the issuer secret trust roots differ) into their own test that doesn't need to re-use the same test functions.

But none of that blocks this PR.  We can come back and clean this up after.

Reply:
Yes that makes total sense. I also agree it will be better if split. Thanks for the tips !